atd-etl/cris_import/cris_import.py

```diff
@@ -0,0 +1,801 @@
+        extract_tmpdir = tempfile.mkdtemp()
+        unzip_command = f'7za -y -p{ZIP_PASSWORD} -o"{extract_tmpdir}" x "{archives_directory}/{filename}"'
+        print(unzip_command)
+        os.system(unzip_command)
+        extracted_csv_directories.append(extract_tmpdir)
+    return extracted_csv_directories
+
+
+def cleanup_temporary_directories(
+    zip_location, pgloader_command_files, extracted_archives
+):
+    """
+    Remove directories that have accumulated during the flow's execution
+
+    Arguments:
+        zip_location: A string containing a path to a temporary directory
+        extracted_archives: A list of strings each containing a path to a temporary directory
+        unit_import_tmpdirs: A list of strings each containing a path to a temporary directory
+        person_import_tmpdirs: A list of strings each containing a path to a temporary directory
+        primaryperson_import_tmpdirs: A list of strings each containing a path to a temporary directory
+        charges_import_tmpdirs: A list of strings each containing a path to a temporary directory
+
+    Returns: None
+    """
+
+    shutil.rmtree(zip_location)
+
+    print(pgloader_command_files)
+    for directory in pgloader_command_files:
+        shutil.rmtree(directory)
+
+    for directory in extracted_archives:
+        shutil.rmtree(directory)
+
+    return None
+
+
+def upload_csv_files_to_s3(extract_directory):
+    """
+    Upload CSV files which came from CRIS exports up to S3 for archival
+
+    Arguments:
+        extract_directory: String denoting the full path of a directory containing extracted CSV files
+
+    Returns:
+        extract_directory: String denoting the full path of a directory containing extracted CSV files
+            NB: The in-and-out unchanged data in this function is more about serializing prefect tasks and less about inter-functional communication
+    """
+
+    session = boto3.Session(
+        aws_access_key_id=AWS_ACCESS_KEY_ID,
+        aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
+    )
+    s3 = session.resource("s3")
+
+    # for extract_directory in extracts:
+    for filename in os.listdir(extract_directory):
+        print("About to upload to s3: " + filename)
+        destination_path = AWS_CSV_ARCHIVE_PATH + "/" + str(datetime.date.today())
+        s3.Bucket(AWS_CSV_ARCHIVE_BUCKET_NAME).upload_file(
+            extract_directory + "/" + filename,
+            destination_path + "/" + filename,
+        )
+    return extract_directory
+
+
+def remove_archives_from_sftp_endpoint(zip_location):
+    """
+    Delete the archives which have been processed from the SFTP endpoint
+
+    Arguments:
+        zip_location: Stringing containing path of a directory containing the zip files downloaded from SFTP endpoint
+
+    Returns: None
+    """
+    with SshKeyTempDir() as key_directory:
+        write_key_to_file(key_directory + "/id_ed25519", SFTP_ENDPOINT_SSH_PRIVATE_KEY + "\n") 
+
```

Comment:
We write out that SSH key again we got from 1password. I found it quicker to write it out again than pass the first location around in a variable down to here.